ArduPlane/mode_takeoff.cpp

```diff
@@ -68,7 +68,9 @@ bool ModeTakeoff::_enter()
 
 void ModeTakeoff::update()
 {
-    if (!takeoff_started) {
+ // don't setup waypoints if we dont have a valid position and home!
+ if (plane.current_loc.initialised() && AP::ahrs().home_is_set()) {
```

Comment:
This would be a less intrusive change if it early returned. 
```
    if (!(plane.current_loc.initialised() && AP::ahrs().home_is_set())) {
        plane.calc_nav_roll();
        plane.calc_nav_pitch();
        SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, 0.0);
        return
    }
```
I think you should/could also add `plane.throttle_suppressed` as a reason to early return. I think currently it will just spam messages in that case. Not your problem of course....